### ISSUE TITLE
Add ability to upload screenshots from lazer

### ISSUE THREAD

**Issue by Game4all** [ppy/osu#10718]
**Describe the new feature:** 

osu!stable has the (secret?) ability to screenshot and directly upload the screenshot on internet with the keybinding Shift+F12 and this feature is quite convenient when sharing screenshots of scores in the game chat / on any messaging app (discord). This basically asks for porting this feature to lazer.

**Proposal designs of the feature:** 

The keybind could be added back to lazer along with the feature. To allow a use on mobile aswell, we could probably add a "share" button on the results screen (possibly on the score card?) that would screenshot and upload it, and do whatever action appropriated to the platform with the screenshot url.

Since osu!web now got a page to see individual scores, the share button could alternatively copy to clipboard / chat the online score url, given that the score was submitted.

**Comment by autoteelar** [comment]
i agree, this feature is one of the absolute MOST used features in the game by almost all players I've ever talked to at least

**Comment by Game4all** [comment]
If anyone hasn't this yet on his radar, I may tackle this but I'll probably require direction for the API uploading routes, assuming of course these are implemented and live.

**Comment by peppy** [comment]
It will need osu-web implementation as a prerequisite.

**Comment by LiquidPL** [comment]
I'm interested in tackling this, what would be the approach for implementing the web-side endpoints? Should I integrate it with whatever infrastructure is already there for screenshots on stable, or should I make it a separate thing?

**Comment by peppy** [comment]
@LiquidPL should be an `osu-web` implementation that roughly matches what stable does. I can provide the code from `web-10` if it helps, just ask.

**Comment by LiquidPL** [comment]
That could be useful yeah, please send it my way whenever you have the time.

**Comment by peppy** [comment]
I've shared with you privately to reduce the overheads in sharing legacy code.

**Comment by Walavouchey** [comment]
just wanna note some additional context in that many tournaments including world cups require use of this feature for proof of achieved score in case of disconnects in the multiplayer match, if there's no stream vod or replay file available

<details><summary>owc ruleset screenshot</summary>

<img width="1293" height="808" alt="image" src="https://github.com/user-attachments/assets/386899e4-df25-4435-a683-26bab59615d7" />

</details>
